parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/internal/PCModuleTestEnv.java

```diff
@@ -23,6 +26,23 @@
 public class PCModuleTestEnv extends PCModule<String, String> {
 
     ModelUtils mu = new ModelUtils(this);
+    Optional<CountDownLatch> workManagerController;
+    private WorkManager<String, String> workManager;
+    private final DynamicLoadFactor limitedDynamicLoadFactor = new LimitedDynamicExtraLoadFactor();
+    private boolean transactionalProducer = true;
+
+    @Override
+    protected DynamicLoadFactor dynamicExtraLoadFactor() {
+        return limitedDynamicLoadFactor;
+    }
+
+    public PCModuleTestEnv(final ParallelConsumerOptions<String, String> optionsInstance,
+                           final boolean transactionalProducer,
+                           final CountDownLatch latch) {
+        super(optionsInstance);
+        this.transactionalProducer = transactionalProducer;
+        this.workManagerController = Optional.of(latch);
+    }
```

Comment:
remove the transactionalProducer flag and corresponding code that was added - in test itself instead only set producer on CommitMode.PERIODIC_TRANSACTIONAL_PRODUCER. 
For the new test - Producer is only needed for the Transactional mode and therefore its enough to not set the producer for other modes rather than create non-transactional producer.

Reply:
Done, makes it lot simpler 👍